src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Collection/IReadOnlySetOfTConverter.cs

```diff
@@ -0,0 +1,49 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+// .NET Framework does not support IReadOnlySet<T>
+#if !NETFRAMEWORK && !NETSTANDARD2_0
```

Comment:
The whole file should be included conditionally. See   `Int128Converter` etc.

Reply:
This will not impact the code to be compiled. It's just a pattern preference that avoid whole-file `ifdef`.

Reply:
Thanks for this comment. I was already confused why we would explicitly add `Compile` properties to the .csproj, but now it makes more sense. With that setup we can easily decide what files should be compiled for specific frameworks, giving a better overview of the compilation process for different frameworks.

I hope you find my added `Condition` in the `csproj` agreeable!